ctapipe/image/reducer.py

```diff
@@ -1,39 +1,109 @@
 """
 Algorithms for the data volume reduction.
 """
-
 from abc import abstractmethod
-
-from ctapipe.core import Component
+import numpy as np
+from ctapipe.image import TailcutsImageCleaner
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import IntTelescopeParameter, BoolTelescopeParameter
+from ctapipe.image.extractor import NeighborPeakWindowSum
+from ctapipe.image.cleaning import dilate
 
 __all__ = [
     'DataVolumeReducer',
     'NullDataVolumeReducer',
+    'TailCutsDataVolumeReducer',
 ]
 
 
-class DataVolumeReducer(Component):
+class DataVolumeReducer(TelescopeComponent):
     """
     Base component for data volume reducers.
     """
 
-    @abstractmethod
-    def __call__(self, waveforms):
+    def __init__(
+        self,
+        config=None,
+        parent=None,
+        subarray=None,
+        image_extractor=None,
+        **kwargs
+    ):
+        """
+        Parameters
+        ----------
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        subarray: ctapipe.instrument.SubarrayDescription
+            Description of the subarray
+        image_extractor: ctapipe.image.extractor.ImageExtractor
+            The ImageExtractor to use for 'TailCutsDataVolumeReducer'.
+            If None, then NeighborPeakWindowSum will be used by default.
+        kwargs
+        """
+
+        self.subarray = subarray
+        super().__init__(config=config, parent=parent, subarray=subarray, **kwargs)
+
+        if image_extractor is None:
+            image_extractor = NeighborPeakWindowSum(parent=self, subarray=subarray)
+        self.image_extractor = image_extractor
+
+        self.cleaner = TailcutsImageCleaner(parent=self, subarray=subarray)
+
+    def __call__(self, waveforms, telid=None, selected_gain_channel=None):
         """
         Call the relevant functions to perform data volume reduction on the
         waveforms.
 
         Parameters
         ----------
-        waveforms : ndarray
+        waveforms: ndarray
             Waveforms stored in a numpy array of shape
             (n_pix, n_samples).
+        telid: int
```

Comment:
This is called `tel_id` everywhere else I think, we should be consistent

Reply:
At least in `extractor.py` and `calibrator.py` it's called telid, that's why i write it like this without `_`.
should i rewrite there too?